wingtips-core/src/main/java/com/nike/wingtips/Tracer.java

```diff
@@ -239,8 +239,11 @@
 
     /**
      * The list of span lifecycle listeners that should be notified when span lifecycle events occur.
+     * Note that we use a {@link CopyOnWriteArrayList} to prevent {@link java.util.ConcurrentModificationException}s
+     * from occurring if spans are being started/completed/etc while this list is being changed. This is deemed
+     * acceptable since iterations over this list happens frequently, but changes to the list should be rare.
      */
-    private final List<SpanLifecycleListener> spanLifecycleListeners = new ArrayList<>();
+    private final List<SpanLifecycleListener> spanLifecycleListeners = new CopyOnWriteArrayList<>();
```

Comment:
curious what's changing this in a way that would result in CME? seems dangerous to change this after config time..

Reply:
The CME is happening during config time. The scenario we're running into is an async process that spins off during app startup that goes to retrieve some quick data w/ an HTTP call, which results in a span. That span is sometimes started or completed at the same time the main area of app startup is adding `SpanLifecycleListener`s, which results in CME.

I agree it's a bit of a smell. There are potentially other ways to solve this, e.g. not allowing async stuff during app startup. But that seems overly restrictive, and ultimately I think gracefully handling concurrent access is worthwhile. 🤷

Reply:
yeah we currently don't allow late changes to the configuration (eg SpanLifecycleListeners are immutable), which is another way. this has the impact that bootstrap config itself is not traced.. but not likely an answer here